File: 090224.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <unordered_set>

using namespace std;

class Solution {
public:
    vector<int> largestDivisibleSubset(vector<int>& nums) {
        vector<int> l;
        for(int i:nums){
            vector<int> d = {i};
            for(int j:nums){
                if (i == j) continue;

                int fit = 1;
                for (int k: d)
                    if(!(k%j == 0 || j%k == 0))
                        fit = 0;
                if(fit)
                    d.push_back(j);
            }
            l = max(d, l, [](vector<int> a, vector<int> b){return a.size() < b.size();});
        }
        return l;
    }
};

int main(){
    Solution s;
    vector<int> nums = {1,2,3};
    vector<int> l = s.largestDivisibleSubset(nums);
    for(int i: l){
        cout << i << " ";
    }
    cout << endl;
    return 0;
}

File: 090224.py
from typing import List


class Solution:
    def largestDivisibleSubset(self, nums: List[int]) -> List[int]:
        l = []
        for i, n in enumerate(nums):
            d = [n]
            for j in nums[i+1:]:
                for k in d:
                    if not (k%j and j%k):
                        d.append(j)
            l = max(d, l, key=lambda x:len(x))
        
        return l
    

s = Solution()
print(s.largestDivisibleSubset([1,2,3]))